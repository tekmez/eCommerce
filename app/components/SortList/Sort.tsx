"use client";
import { Card, Radio, Space } from "antd";
import { usePathname, useRouter, useSearchParams } from "next/navigation";
import type { RadioChangeEvent } from "antd";
const Sort = () => {
  const router = useRouter();
  const pathname = usePathname();
  const searchParams = useSearchParams();
  const createPageURL = (sort: string) => {
    const params = new URLSearchParams(searchParams);
    params.delete("sortBy");
    params.set("sortBy", sort);
    return `${pathname}?${params}`;
  };

  const onChange = (e: RadioChangeEvent) => {
    const newUrl = createPageURL(e.target.value);
    router.push(newUrl);
  };
  return (
    <Card title="Sort By" style={{ width: 250 }}>
      <Radio.Group onChange={onChange}>
        <Space direction="vertical">
          <Radio value={"price"}>Price low to High</Radio>
          <Radio value={"createdAt"}>Old to new</Radio>
        </Space>
      </Radio.Group>
    </Card>
  );
};

export default Sort;
