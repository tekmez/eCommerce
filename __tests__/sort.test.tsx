import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Sort from "../app/components/SortList/Sort";
import { useRouter } from "next/navigation";
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn().mockReturnValue({
    push: jest.fn(),
  }),
  usePathname: jest.fn().mockReturnValue("/mock-pathname"),
  useSearchParams: jest.fn().mockReturnValue(new URLSearchParams("")),
}));

describe("Sort component", () => {
  test("snapshot test", () => {
    const { container } = render(<Sort />);
    expect(container).toMatchSnapshot();
  });
  test("renders Sort component correctly", () => {
    const { getByLabelText } = render(<Sort />);
    const priceRadioButton = getByLabelText("Price low to High");

    // Simulate radio button change
    fireEvent.click(priceRadioButton);

    // Check if useRouter is called with the correct URL
    expect(useRouter().push).toHaveBeenCalledWith(
      "/mock-pathname?sortBy=price"
    );
  });
});
