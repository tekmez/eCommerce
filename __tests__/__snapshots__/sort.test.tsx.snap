// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort component snapshot test 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-1okl62o"
    style="width: 250px;"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Sort By
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-radio-group ant-radio-group-outline css-dev-only-do-not-override-1okl62o"
      >
        <div
          class="ant-space css-dev-only-do-not-override-1okl62o ant-space-vertical ant-space-gap-row-small ant-space-gap-col-small"
        >
          <div
            class="ant-space-item"
          >
            <label
              class="ant-radio-wrapper css-dev-only-do-not-override-1okl62o"
            >
              <span
                class="ant-radio ant-wave-target"
              >
                <input
                  class="ant-radio-input"
                  type="radio"
                  value="price"
                />
                <span
                  class="ant-radio-inner"
                />
              </span>
              <span>
                Price low to High
              </span>
            </label>
          </div>
          <div
            class="ant-space-item"
          >
            <label
              class="ant-radio-wrapper css-dev-only-do-not-override-1okl62o"
            >
              <span
                class="ant-radio ant-wave-target"
              >
                <input
                  class="ant-radio-input"
                  type="radio"
                  value="createdAt"
                />
                <span
                  class="ant-radio-inner"
                />
              </span>
              <span>
                Old to new
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
